test/extensions/filters/http/lua/lua_filter_test.cc

```diff
@@ -1800,6 +1800,40 @@ TEST_F(LuaHttpFilterTest, GetRequestedServerName) {
   EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_->decodeHeaders(request_headers, true));
 }
 
+// Verify that binary values could also be extracted from dynamicMetadata() in LUA filter.
+TEST_F(LuaHttpFilterTest, GetDynamicMetadataBinaryData) {
+  const std::string SCRIPT{R"EOF(
+    function envoy_on_request(request_handle)
+      local metadata = request_handle:streamInfo():dynamicMetadata():get("envoy.pp")
+      local bin_data = metadata["bin_data"]
+      local data_length = string.len(metadata["bin_data"])
+      for idx = 1, data_length do
```

Comment:
I posted a comment but can't find it, so sorry if this shows up as a duplicate:

Make your Lua something like:
bin_data = "he\0llo"
escaped = ""
for idx = 1, string.len(bin_data) do
  escaped = escaped .. string.format("\\x%02x", string.byte(bin_data, idx))
end

Then you can have a single EXPECT_CALL like:
// Hex values for "he\0llo"
EXPECT_CALL(*filter_, scriptLog(spdlog::level::trace, StrEq(R"Hex Data: \x68\x65\x00\x6c\x6c\x6f")));